### PR TITLE
Sadikov Damir. Lab 4. Var 3

### PR DESCRIPTION
**Pull request by Damlrca**
Lab 4.
Var 3

### CODE REVIEW

File: mlir/lib/Transforms/lab4/sadikov_damir/SadikovMLIRDepthPass.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    : public PassWrapper<SadikovMLIRDepthPass, OperationPass<func::FuncOp>> {
+public:
+  StringRef getArgument() const final { return "sadikov-mlir-depth-pass"; }
+  StringRef getDescription() const final {
+    return "sadikov-mlir-depth-pass : pass that counts the max depth of region "
+           "nests in the function";
+  }
+  void runOnOperation() override {
+    Operation *func = getOperation();
+    func->setAttr("depth_of_region",
```
Comment (aobolensk) [inline mlir/lib/Transforms/lab4/sadikov_damir/SadikovMLIRDepthPass.cpp]: Why depth **of region**?

File: mlir/test/Transforms/lab4/sadikov_damir/test.mlir [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+}
+
+//--- func3.mlir
+module{ 
+    // CHECK: func.func @func3(%arg0: memref<1024xf32>, %arg1: index, %arg2: index, %arg3: index) -> f32 attributes {depth_of_region = 3 : i32} {
+    func.func @func3(%buffer: memref<1024xf32>, %lb: index, %ub: index, %step: index) -> f32 {
+        %sum_0 = arith.constant 0.0 : f32
+        %c0 = arith.constant 0.0 : f32
+        %sum = scf.for %iv = %lb to %ub step %step
```
Comment (aobolensk) [inline mlir/test/Transforms/lab4/sadikov_damir/test.mlir]: Check on something different, like scf.while+scf.if+...

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: mlir/lib/Transforms/lab4/sadikov_damir/SadikovMLIRDepthPass.cpp (hunk 1 above)
Comment (Damlrca) [reply mlir/lib/Transforms/lab4/sadikov_damir/SadikovMLIRDepthPass.cpp]: Just accidentally named it in wrong way. I think depth_of_func is more correct. Renamed it like this

Comment (aobolensk) [reply mlir/lib/Transforms/lab4/sadikov_damir/SadikovMLIRDepthPass.cpp]: Ok

File: mlir/test/Transforms/lab4/sadikov_damir/test.mlir (hunk 2 above)
Comment (Damlrca) [reply mlir/test/Transforms/lab4/sadikov_damir/test.mlir]: Сhanged for loop to while loop in this test

**Review by aobolensk — APPROVED** [review]

**Pull request merged**